### PR TITLE
ENT-5258 db schema set-up only via command line flag

### PR DESCRIPTION
**Pull request by blsemo**
Removing the ability to initialise schema from the node config, and add a new sub-command to initialise the schema (that does not do anything else and exits afterwards).
Also adding a command line flag that allow app schema to be maintained by hibernate for legacy cordapps, tests or rapid development.
Patching up mock net and driver test frameworks so they create the required schemas for tests to work.
Modified network bootstrapper to run an initial schema set-up so it can register nodes.

### CODE REVIEW

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/DriverDSLImpl.kt [hunk 1]
```diff
@@ -139,7 +147,8 @@ class DriverDSLImpl(
         val cordappsForAllNodes: Collection<TestCordappInternal>?,
         val djvmBootstrapSource: Path?,
         val djvmCordaSource: List<Path>,
-        val environmentVariables : Map<String, String>
+        val environmentVariables : Map<String, String>,
+        val allowHibernateToManageAppSchema: Boolean = false
```
Comment (lankydan) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/DriverDSLImpl.kt]: Does a default of `false` make sense? If I understand this correctly, if it is `true` then it works as it does already. Therefore it should default to `true` so developers don't have to add this to all of their tests unless they want to change the existing behaviour.

**Review by mnesbit — COMMENTED** [review]
For driver tests what is ratio of allowHibernateToManageAppSchema = true to allowHibernateToManageAppSchema = false ? Just wondering what correct default is as this clearly touches a lot of tests

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/DriverDSLImpl.kt (hunk 1 above)
Comment (mnesbit) [reply testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/DriverDSLImpl.kt]: Yes I have same question. When we turn off hibernate what is the alternative? Does liquibase get applied correctly? I get that we might want to 'encourage' people to do the right thing first time, just trying to work out if this is a bit early in the cordapp dev maturity pathway, or is a one shot pain to sort out lurking problems.

Comment (blsemo) [reply testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/DriverDSLImpl.kt]: In the long run, I definitely want this to default to false! You shouldn't rely on hibernate sorting out schema glitches unless you explicitly opt in to get that. The default behaviour should be that an integration test also tests your schema migration!
In the short term, a default of `true` until our test CorDapp (that gets implicitly loaded even if not used in most driver tests) is sorted out and we can be sure that schema migration for CorDapps just works if the schema migration scripts are present and correct.

**Review by mnesbit — DISMISSED** [review]
OK

**Review by mnesbit — DISMISSED** [review]
OK

**Pull request closed without merging**